### PR TITLE
GlusterFS: Use custom StorageClass for S3

### PR DESCRIPTION
**Pull request by jarrpa**
Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1551426

Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

### CODE REVIEW

**Comment by jarrpa** [comment]
@sdodson @michaelgugino I'm running into an issue where this is not working in my test setup. Is there any reason a StorageClass with a URL to a Service would not be able to talk to that Service over its Endpoints address? Is this a fundamentally broken thing or something wrong with my networking? The part that has me really confused is that this seems to work fine in Kubernetes... something to do with the SDN?

**Comment by sdodson** [comment]
So, which host is going to try to access `resturl: "http://{{ glusterfs_heketi_url }}:{{ glusterfs_heketi_port }}"` ? that's going to be http://heketi.gluster.svc.cluster.local:8080 ? can you verify the service actually has endpoints?

**Comment by jarrpa** [comment]
* Which host: No idea, really. It's whichever host is running the in-tree GlusterFS provisioner, which I guess would be one of the masters?
* Yes, something like that URL.
* Yes, the service has endpoints.

To those points, I tried to curl the Service URL from a master node with no luck, no route to host. I then tried to use the Endpoints IP, which is what we do for Kubernetes, and that didn't work in the StorageClass nor could I curl the address either. However, if I mark the master node as schedulable it seems to be included in the SDN because I can then curl the Endpoints IP. The StorageClass still fails, though, even in that scenario.

**Comment by jarrpa** [comment]
@sdodson @michaelgugino I managed to figure out the problem with my setup: my dnsmasq configuration was interfering with the openshift-sdn configuration. I can verify that this works. Please review and LGTM if it's true. :)

**Comment by sdodson** [comment]
@ewolinetz is this a known logging problem?

**Comment by jarrpa** [comment]
@sdodson Updated this PR to account for #7729. If we hit the logging test failure again could you waive it?

**Comment by ewolinetz** [comment]
@sdodson I didn't get a chance to see the output from the logging test before it was rerun... I'll check if it fails again

**Comment by sdodson** [comment]
/lgtm
@ewolinetz https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/7405/test_pull_request_openshift_ansible_logging/7354/

File: roles/openshift_storage_glusterfs/tasks/gluster_s3_deploy.yml [hunk 1]
```diff
@@ -111,3 +156,15 @@
...
   retries: "{{ (glusterfs_timeout | int / 10) | int }}"
+
+- name: Delete GlusterFS StorageClass for S3
+  oc_obj:
+    namespace: "{{ glusterfs_namespace }}"
+    kind: "{{ item.kind }}"
+    name: "{{ item.name | default(omit) }}"
+    selector: "{{ item.selector | default(omit) }}"
+    state: absent
+  with_items:
```
Comment (michaelgugino) [inline roles/openshift_storage_glusterfs/tasks/gluster_s3_deploy.yml]: No apparent usage for with_items here.

**Comment by ewolinetz** [comment]
@sdodson thanks, no that is a flake.

File: roles/openshift_storage_glusterfs/tasks/gluster_s3_deploy.yml (hunk 1 above)
Comment (jarrpa) [reply roles/openshift_storage_glusterfs/tasks/gluster_s3_deploy.yml]: Holdover from a pattern I copy/pasted. Fixing.

**Comment by jarrpa** [comment]
@sdodson Changes made to address @michaelgugino's comments. Please LGTM again.

**Comment by jarrpa** [comment]
I lied. NOW I pushed the requested changes.

**Pull request merged**